text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+    * **Application Rework:** Smart contracts (chaincode) would need to be re-implemented or adapted to the Fabric-X programming model (e.g., as 'Flows' using FSC).
+* **Potential for Easing Transition (Future Iterations):**
+    * **Traditional Chaincode Support:** To lower the barrier for existing applications, future iterations of Fabric-X could explore providing a compatibility layer or environment within the endorser services to support the execution of traditional Fabric chaincode (shim interface). This could potentially be achieved by leveraging Fabric's "chaincode as a service" (CCaaS) builder to support remote chaincodes.
+    * **Alternative PDC Implementation:** An alternative implementation of Private Data Collections (PDCs) could be introduced, potentially leveraging the endorser's local database in conjunction with the main ledger for hash commitments, tailored to Fabric-X's architecture.
+    * **Multiple Channel Support:** While the current Fabric-X implementation operates on a single-channel concept (or a global namespace) for simplicity and to maximize performance in its target use cases (like CBDCs which often imply a single logical ledger), supporting multiple channels or equivalent data partitioning mechanisms is architecturally feasible and could be incorporated if required.
+    * The introduction of support for traditional chaincode, refined PDCs, and multiple channels would be subject to separate, future RFCs once the new governance model for coexisting Fabric implementations is established and the core Fabric-X platform has stabilized.
+* **Coexistence and Targeted Adoption:**
+    * It is crucial to understand that **migration to Fabric-X is not intended or necessary for all existing Fabric networks.** Fabric (Classic) is a mature, stable platform that effectively serves a vast number of production use cases.
+    * Fabric-X is primarily aimed at **new, demanding use cases** that require performance levels significantly beyond what current Fabric can offer (e.g., CBDCs, high-frequency digital asset exchanges).
+    * Organizations should only consider Fabric-X if their performance requirements genuinely necessitate it. Otherwise, they would face the **unnecessary operational complexity** of a distributed microservice deployment and a potentially involved migration process without fully benefiting from its core advantage—extreme performance. Most existing Fabric use cases are well-served by its current performance capabilities.
+    * The vision is for Fabric and Fabric-X to **coexist within the broader Hyperledger Fabric ecosystem**, each serving different segments and requirements. Fabric-X expands the ecosystem's capabilities, rather than seeking to universally replace the established Fabric implementation.
```

Comment:
Once again, failing to understand why Fabric-X cannot just be a standalone Hyperledger project, rather than pushing the fragmentation of the Fabric ecosystem.